tests/test_bucket_versioning.py

```diff
@@ -0,0 +1,396 @@
+import logging
+import random
+from framework.customizations.marks import tier1
+
+log = logging.getLogger(__name__)
+
+
+class TestBucketVersioningOperations:
+    """
+    Test S3 Bucket versioning operations using NSFS
+    """
+
+    obj_name = "obj_1"
+    obj_data = "Original data"
+
+    def setup_versioned_bucket(self, c_scope_s3client):
+        """
+        Setup the versioned bucket
+
+        Args:
+            c_scope_s3client (Object):
+        Return:
+            string: Versioned Bucket name
+        """
+        # Create regular bucket
+        bucket = c_scope_s3client.create_bucket()
+
+        # enable bucket versioning
+        c_scope_s3client.put_bucket_versioning(bucket, status="Enabled")
+        log.info(f"Enabled bucket versioning on bucket {bucket}")
+
+        return bucket
+
+    def list_all_versions_of_the_object(
+        self, c_scope_s3client, bucket_name, object_name
+    ):
+        """
```

Comment:
This logic might be useful in future tests outside of this suite - please relocate this to be a method of the `S3Client` class

Reply:
Since we are implementing all boto3 API calls in s3Client class, I have implemented `list_object_versions` method there and the purpose of `list_all_versions_of_the_object` method is to filter out versions based on keys which is again depend on `list_object_versions` API call

Reply:
I understand - but this specific use of `list_object_versions` is the most trivial and likely to repeat in files other than this one. Whether you place it under `S3Client` or a new `s3_utils` file is less the point - I just don't want to copy-paste it in the future 🤷

Reply:
I moved this logic to utils file for reusability purpose

Reply:
New pass log link: 
https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-noobaa-sa/280/console